### DISCUSSION TITLE
Using scopes in attach action

### DISCUSSION THREAD

**Discussion by gregoryloichot** [filamentphp/filament#13345] [Help] [answered]
### Package

Actions

### Package Version

v3.2

### How can we help you?

Hello, 

I have an AttachAction in a RelationManager. Let say I want to attach a Tag to an Article. 
When I attach a Tag, I would like to apply a scope (like Tag::myScope()), to filter the options in the Selectbox. How can I do this ?

**Comment by leandrocfe** [comment, marked as answer]
https://filamentphp.com/docs/3.x/panels/resources/relation-managers#scoping-the-options-to-attach

**Comment by gregoryloichot** [comment]
Yes thanks, I saw it but did not figure out how to make it work. Finally it was more a Laravel related issue, sorry

But let me post the example:

```php
Tables\Actions\AttachAction::make()
                    ->label('Add manager')
                    ->preloadRecordSelect()
                    ->form(fn (AttachAction $action): array => [
                        $action->getRecordSelect(),
                        Forms\Components\TextInput::make('role_id')->hidden()->default('3')->required(),
                    ])
                    ->recordSelectOptionsQuery(fn (Builder $query) => $query->whereHas('roles', function (Builder $query) {
                        $query->where('code', '=', Role::admin()->code);
                    })),
```

Might I ask one more question about the form method? This action write in a pivot table with `user_id`, `role_id` and `tp_id`. The fields `user_id` and `tp_ip` are correctly filled by the action. However, `role_id` is not and I want it to be filled with a constant value. The code above has no effect. How could I achieve this?

(I want the field to be hidden ; it works if it is not)

**Reply by leandrocfe** [reply]
try `mutateFormDataUsing` instead of hidden field
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#customizing-data-before-saving
